src/Cabal2nix.hs

```diff
@@ -99,19 +102,51 @@ readP p = eitherReader $ \s -> case [ r' | (r',"") <- P.readP_to_S p s ] of
                                     (r:_) -> Right r
                                     _     -> Left ("invalid value " ++ show s)
 
-parsePlatform :: P.ReadP r Platform
-parsePlatform = do arch <- P.choice [ P.string "i686" >> return I386
-                                    , P.string "x86_64" >> return X86_64
-                                    , P.string "armv7l" >> return Arm
-                                    , P.string "armv6l" >> return Arm
-                                    -- Not yet supported yet, see Cabal #5221
-                                    --, P.string "aarch64" >> return AArch64
-                                    , P.string "mipsel" >> return Mips
-                                    ]
-                   _ <- P.char '-'
-                   os <- P.choice [ P.string "linux" >> return Linux, P.string "darwin" >> return OSX
-                                  , P.string "windows" >> return Windows]
-                   return (Platform arch os)
+-- | Replicate the normalization performed by GHC_CONVERT_CPU in GHC's aclocal.m4
+-- since the output of that is what Cabal parses.
+ghcConvertArch :: String -> String
+ghcConvertArch = \case
```

Comment:
Is the use of LambdaCase here really necessary? It feels like this code could easily have been written with normal pattern matching and it would look almost exactly the same.

Reply:
I generally prefer to not duplicate identifiers unnecessarily - especially with pattern guards that apply to the thing being cased on: `case foo of { foo' | foo' == bar -> ... }` or `case foo of { _ | foo == bar -> ... }`?

Happy to change the style if you want it the other way though. :)

Reply:
I would prefer the conventional style, yes. I believe that it can be written quite straight-forwardly without duplicating any identifiers (although I haven't actually tried it so I might be wrong).

Reply:
Done with use of an underscore.